depthai_nodes/ml/parsers/classification.py

```diff
@@ -31,21 +32,57 @@ class ClassificationParser(dai.node.ThreadedHostNode):
          An object with attributes `classes` and `scores`. `classes` is a list of classes, sorted in descending order of scores. `scores` is a list of corresponding scores.
     """
 
-    def __init__(self, classes: List[str] = None, is_softmax: bool = True):
+    def __init__(self):
         """Initializes the ClassificationParser node.
 
-        @param classes: List of class names to be used for linking with their respective
-            scores.
-        @param is_softmax: If False, the scores are converted to probabilities using
-            softmax function.
+        Attributes
+        ----------
+        archive_metadata : list
+            List of all archive head objects: [<depthai.nn_archive.v1.Head object>, ...]
+        head_name : str
+            If the list has multiple heads, this will specify which head to use. will throw an error if archive heads is longer then 1 and no head name is provided
         """
+        super().__init__()
+        self.output_layer_name: str = ""
+        self.classes: List = None
+        self.n_classes: int = 0
+        self.is_softmax: bool = True
+
+    def build(
+        self,
+        head_metadata: Union[List, Dict],
+        head_name: str = "",
+        is_softmax: bool = True,
+    ):
+        super().build(head_metadata, head_name)
+
+        ## should we add or not?
+        # if (self.head_configs["parser"] != "ClassificationParser"):
+        #     raise ValueError("Head is not a classification head, please correct the nn_archive")
+
+        output_layers = self.head_configs["outputs"]
+        if len(output_layers) != 1:
+            raise ValueError(
+                "Only one output layer supported for classification, please correct nn_archive/ model"
+            )
+        self.output_layer_name = output_layers[0]
+
+        try:
+            self.classes = self.head_configs["classes"]
+        except KeyError:
+            print(
+                "No classes provided in nn_archive metadata. Please provide class names in the nn_archive"
+            )
+
+        try:
+            self.n_classes = self.head_configs["n_classes"]
+        except KeyError:
+            print(
+                "No n_classes provided in nn_archive metadata. Please provide number of classes in the nn_archive"
+            )
```

Comment:
Are these OK KeyErrors? Or should they be adjusted?

Reply:
Hm, This does not error out, right? But just prints out the warning. So, `setClasses` method should be used?